src/collects/seashell/backend/template.rkt

```diff
@@ -55,8 +55,14 @@
     [(url-string? source)
      (define surl (string->url source))
      (cond
-       [(equal? (url-scheme surl) "file")
-        (call/input-url surl get-pure-port thunk)]
+       [(string=? (url-scheme surl) "file")
+         (call/input-url surl get-pure-port thunk)]
+       [(string=? (url-scheme surl) "ssh")
+         (match-define (list _ host file) (regexp-match #rx"//(.*@[^:]*):(.*)" source))
+         (define-values (sshproc sshout sshin ssherr)
+           (subprocess #f #f #f (read-config 'ssh-binary)
+             host (string-append "cat " file)))
+         (thunk sshout)]
```

Comment:
Make sure you close sshout/ssherr and you use dynamic-wind (see below) to close sshout when thunk returns.